### ISSUE TITLE
'type != null || const Object() is! T': GetIt: The compiler could not infer the type. You have to provide a type and optional a name.

### ISSUE THREAD

**Issue by jnknewaj** [fluttercommunity/get_it#134]
At the launch of my app, I get the following error :

_AssertionError ('package:get_it/get_it_impl.dart': Failed assertion: line 274 pos 7: 'type != null || const Object() is! T': GetIt: The compiler could not infer the type. You have to provide a type and optional a name. Did you accidentally do  `var sl=GetIt.instance();` instead of var sl=GetIt.instance;)

I can assure that I didn't do **GetIt.instance()**, rather I did **GetIt.instance**.

Here is how I configured `getIt`:

```
final GetIt getIt = GetIt.instance;

@injectableInit
void configureInjection(String env) {
  $initGetIt(getIt, environment: env);
}
```
And then : 

```
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp();
  configureInjection(Environment.prod);
  runApp(MyApp());
}
```

**Comment by escamoteur** [comment]
I honestly have no idea what exactly Injectable does. Maybe you haven't the latest version of injectable? Otherwise please open an issue there, thanks.

**Comment by arslan9380** [comment]
facing the same issue on latest version of everything. any solution so far?

**Comment by Tristan-2021** [comment]
Hello people, I had the same problem there is the dependencies injection and resolved it. As you say here "The Compiler Could not Infer The Type.", It refers that the compiler can not guess what type you are injecting ..
    
![enferir _type](https://user-images.githubusercontent.com/79274889/163486043-018b774f-ac71-4ffa-8663-706a2937b8a2.png)

You always have to add the guy, it helps that get_it can recognize which is kind as of each constructor
![enferir _type](https://user-images.githubusercontent.com/79274889/163487359-fb759223-b2c1-479a-b865-19e05d1a1248.png)

**Comment by luangs7** [comment]
same problem here

**Comment by MuhammadAmin-Ibrohimov** [comment]
I also have same problem! Please help meee!

**Comment by escamoteur** [comment]
Please provide the code where it happens otherwise I can't help you.
Am 3. Apr. 2024, 19:37 +0200 schrieb MuhammadAmin Ibrohimov ***@***.***>:

**Comment by over01470914** [comment]
I encountered this issue today, sharing my problems as follow:
I use GetIt.instance.registerFactory to register my service, but my factory type is a abstract interface. I just use a implemented subclass inheritted from that abstract interface and it's solved, like:

```
abstract interface class A{
}

class B implements A{
}

class C{
  C(required this.a);
  final A a;
}
```

originally, I use:

```
getIt.registerFactory<A>(
  ()=>A()
)
getIt.registerFactory<C>(
  ()=>C(getIt())
)
```

but it continuely gives me errors, then I tried:
```
getIt.registerFactory<B>(
  ()=>B()
)
getIt.registerFactory<C>(
  ()=>C(getIt.get<B>())
)
```
then it worked! I think this API might not accept a abstract class, it must be given by a specific refering class?
